src/+otp/+constrainedpendulum/ConstrainedPendulumProblem.m

```diff
@@ -0,0 +1,35 @@
+classdef ConstrainedPendulumProblem < otp.Problem
+    %CONSTRAINEDPENDULUM PROBLEM This is a Hessenberg Index-2 DAE problem posed in terms of three constraints
+    %
+    
+    properties
+        Constraints
+        ConstraintsJacobian
+        Z0 = [0; 0; 0];
+    end
+
+    methods
+        function obj = ConstrainedPendulumProblem(timeSpan, y0, parameters)
+            obj@otp.Problem('Constrained Pendulum', 4, timeSpan, y0, parameters);
+        end
+    end
+    
+    methods (Access = protected)
+        function onSettingsChanged(obj)
+            m = obj.Parameters.Mass;
+            l = obj.Parameters.Length;
+            g = obj.Parameters.Gravity;
+
+            % get initial energy
+            initialconstraints = otp.constrainedpendulum.constraints([], obj.Y0, g, m, l, 0);
+            E0 = initialconstraints(3);
+            
+            obj.RHS = otp.RHS(@(t, y) otp.constrainedpendulum.f(t, y, g, m, l, E0));
```

Comment:
This RHS will lead to confusion as a call to `solve` will not enforce constraints, right?  Ideally, `RHS` should be in a form that a built-in solver can integrate, but we can let that slide for an index-2 DAE.  I suggest the default RHS should use a mass matrix and an `f` that operates on the full state (diff and alg).  Then there can be a `RHSDifferential` and `RHSAlgebraic` to get the individual pieces.

Reply:
Also it looks like `'Vectorized', 'on'` can be added to RHS